lib/reverse_sentence.rb

```diff
@@ -1,6 +1,136 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(2n) The array is reviewed twice, once for spaces to populate the space array, and once more piecewise to do all the swaps
+# Space complexity: O(1)
+
+require'pry'
+
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  
+  # short circuit if empty sentence
+  if my_sentence == nil || !my_sentence.include?(" ")
+    return my_sentence
+  end
+  
+  # determines the last index number that will be handled
+  last_index = my_sentence.length - 1 
+  
+  # reverses all the characters in the string
+  i = 0
+  my_sentence[i]
+  half_swap_limit = last_index / 2  + 1
+  
+  until i == half_swap_limit
+    my_sentence[i], my_sentence[last_index - i] = my_sentence[last_index - i], my_sentence[i]
+    
+    i += 1
```

Comment:
Notice that you reverse things a few times, this could be dryed up with a helper method.